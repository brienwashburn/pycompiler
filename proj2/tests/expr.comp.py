[(x,y) for x in foo(4) if x > 0 if (2+3) == 5 for a,b,*c in bar(10) for *d,e,*f in bar(8)]
((x,y) for x in foo(4) if x > 0 if (2+3) == 5 for a,b,*c in bar(10) for *d,e,*f in bar(8))
{(x,y) for x in foo(4) if x > 0 if (2+3) == 5 for a,b,*c in bar(10) for *d,e,*f in bar(8)}
{x:y for x in foo(4) if x > 0 if (2+3) == 5 for a,b,*c in bar(10) for *d,e,*f in bar(8)}
