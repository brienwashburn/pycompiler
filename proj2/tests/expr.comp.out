(Module
 (Expr
  (ListComp
   (Tuple (Name x) (Name y))
   (for
    (Name x)
    in
    (Call
     (func (Name foo))
     (args (Num 4))
     (keywords)
     (starargs #f)
     (kwargs #f))
    if
    (Compare (left (Name x)) (ops Gt) (comparators (Num 0)))
    (Compare
     (left (BinOp (Num 2) Add (Num 3)))
     (ops Eq)
     (comparators (Num 5))))
   (for
    (Tuple (Name a) (Name b) (Starred (Name c)))
    in
    (Call
     (func (Name bar))
     (args (Num 10))
     (keywords)
     (starargs #f)
     (kwargs #f))
    if)
   (for
    (Tuple (Starred (Name d)) (Name e) (Starred (Name f)))
    in
    (Call
     (func (Name bar))
     (args (Num 8))
     (keywords)
     (starargs #f)
     (kwargs #f))
    if)))
 (Expr
  (GeneratorExp
   (Tuple (Name x) (Name y))
   (for
    (Name x)
    in
    (Call
     (func (Name foo))
     (args (Num 4))
     (keywords)
     (starargs #f)
     (kwargs #f))
    if
    (Compare (left (Name x)) (ops Gt) (comparators (Num 0)))
    (Compare
     (left (BinOp (Num 2) Add (Num 3)))
     (ops Eq)
     (comparators (Num 5))))
   (for
    (Tuple (Name a) (Name b) (Starred (Name c)))
    in
    (Call
     (func (Name bar))
     (args (Num 10))
     (keywords)
     (starargs #f)
     (kwargs #f))
    if)
   (for
    (Tuple (Starred (Name d)) (Name e) (Starred (Name f)))
    in
    (Call
     (func (Name bar))
     (args (Num 8))
     (keywords)
     (starargs #f)
     (kwargs #f))
    if)))
 (Expr
  (SetComp
   (Tuple (Name x) (Name y))
   (for
    (Name x)
    in
    (Call
     (func (Name foo))
     (args (Num 4))
     (keywords)
     (starargs #f)
     (kwargs #f))
    if
    (Compare (left (Name x)) (ops Gt) (comparators (Num 0)))
    (Compare
     (left (BinOp (Num 2) Add (Num 3)))
     (ops Eq)
     (comparators (Num 5))))
   (for
    (Tuple (Name a) (Name b) (Starred (Name c)))
    in
    (Call
     (func (Name bar))
     (args (Num 10))
     (keywords)
     (starargs #f)
     (kwargs #f))
    if)
   (for
    (Tuple (Starred (Name d)) (Name e) (Starred (Name f)))
    in
    (Call
     (func (Name bar))
     (args (Num 8))
     (keywords)
     (starargs #f)
     (kwargs #f))
    if)))
 (Expr
  (DictComp
   (Name x)
   (Name y)
   (for
    (Name x)
    in
    (Call
     (func (Name foo))
     (args (Num 4))
     (keywords)
     (starargs #f)
     (kwargs #f))
    if
    (Compare (left (Name x)) (ops Gt) (comparators (Num 0)))
    (Compare
     (left (BinOp (Num 2) Add (Num 3)))
     (ops Eq)
     (comparators (Num 5))))
   (for
    (Tuple (Name a) (Name b) (Starred (Name c)))
    in
    (Call
     (func (Name bar))
     (args (Num 10))
     (keywords)
     (starargs #f)
     (kwargs #f))
    if)
   (for
    (Tuple (Starred (Name d)) (Name e) (Starred (Name f)))
    in
    (Call
     (func (Name bar))
     (args (Num 8))
     (keywords)
     (starargs #f)
     (kwargs #f))
    if))))
